File: store/src/main/java/com/min/store/product/domain/ProductDomain.java
package com.min.store.product.domain;

import lombok.*;

@Getter
public class ProductDomain {

    private final Long id;
    private final String name;
    private final Long price;

    @Builder
    public ProductDomain(Long id, String name, Long price) {
        this.id = id;
        this.name = name;
        this.price = price;
    }

    public static ProductDomain from(String name, Long price) {
        return ProductDomain.builder()
                .name(name)
                .price(price)
                .build();
    }

    public ProductDomain update(ProductUpdateDomain productUpdate) {
        return ProductDomain.builder()
                .id(id)
                .name(productUpdate.getName())
                .price(productUpdate.getPrice())
                .build();
    }

}

File: store/store-product-api/src/main/kotlin/com/store/product/api/domain/ProductDomain.kt
package com.store.product.api.domain

class ProductDomain (
    val id: Long?=null,
    val name: String,
    val price: Long
    ) {

    companion object {
        fun from(name: String, price: Long): ProductDomain {
            return ProductDomain(name = name, price = price)
        }
    }

    fun update(productUpdate: ProductUpdateDomain): ProductDomain {
        return ProductDomain(id, productUpdate.name, productUpdate.price)
    }
}
